### ISSUE TITLE
[SVT_AV1] Open Dev Meeting Jan8-2020

### ISSUE THREAD

**Issue by hassount** [OpenVisualCloud/SVT-AV1#932]
Happy New Year to all !

The second Open Dev Team is scheduled for January 8th 2020, @ 8:30 - 9:30 am PST

Created a new issue to track the discussion items for this meeting, please let us know your thoughts / suggestions for topics on the Agenda.

_Originally posted by @hassount in https://github.com/OpenVisualCloud/SVT-AV1/issues/811#issuecomment-559184213_

### Using the same dialup info from last time:

Join Hangouts Meet
meet.google.com/rty-mkxi-wvi

Meeting ID
https://meet.google.com/rty-mkxi-wvi?hs=122

Phone Numbers
(‪US‬)  +1 929-269-1840‬
PIN: ‪542 747#‬

More phone numbers
https://meet.google.com/tel/rty-mkxi-wvi?pin=9055644317111&hs=1

**Comment by grill05** [comment]
an update on rate-control would be great.

**Comment by luigi311** [comment]
I was playing around with coz profiler and ran the following command 1000 times over night.
`./SvtAv1EncApp -i ../../../Chim*.y4m -enc-mode 4 -b ../../multisocket.mkv`

I told the coz profiler that a unit of work is complete when it gets to EbAppMain.c:L231 after it checks to see if all channels are inactive since what i assumed was that while loop keeps looping after a single frame has been encoded and will keep looping until all the frames are done. If that is incorrect let me know where you want me to place the mark and i can run the test again.

Coz is a Causal Profiling tool that will tell you what to expect if you optimize a specific function. Say optimizing function 1 will result in a 10% performance increase if you optimize the function to be 40% faster but optimizing function 2 will result in a 50% performance increase by optimizing the function by only 10%. 

If you are curious here is the github link to the project https://github.com/plasma-umass/coz that also has a link to nice youtube video where they do a presentation on it.

I have also attached my profiler.coz file so you can use it on the webviewer https://plasma-umass.org/coz/


[profilecoz.zip](https://github.com/OpenVisualCloud/SVT-AV1/files/4028025/profilecoz.zip)

**Comment by luigi311** [comment]
Nice to see a lot of pull requests being closed. Good job on getting people on those! Very encouraging putting in a pull requests and receiving feedback in a timely manner, keep it up :)

**Comment by ePirat** [comment]
As mentioned in #873 I've encountered some problems adding unit tests to CI, so it would be great if we could discuss this.

**Comment by hassount** [comment]
Hello all,

Thank you for joining today, below are some minutes I have captured on what we have discussed. Please let me know if I have missed anything.


**Question1:**
- Look into -film-grain switch does not seem to be working [will check this] [AR]

**Question2:**
- CI takes too long: single threaded
group the tests that can be run in parallel together and create blocks
need to figure out why some of them are taking too long to run

**Question3:**
- API work - what to expect:
cleanup the api and make it future proof by using a dictionary 

**Question4:**
- CI :
need to add encode / decode tests
we might need to use docker images to get the decoders so that we don't have to rebuild them
can host docker images and any item related to testing in https://github.com/OpenVisualCloud/SVT-AV1-Resources

**Question5:**
- how to test same output
- objective-1-fast for speed and BDR differences should be small

**Question6:**
- Status of svt-av1 decoder:
svt-av1 decoder was built for research purposes and the encoder / decoder / common separation is there to make sure it's friendly for implementing non-av1 features in the future.

**Question7:**
- Status of documentation:
Documentation will be pushed to the wiki by next week in time for 0.8.1

**Question8:**
- Macos compiler versions 
one compiler version is enough ? yes

**Question9:**
- Any planned support for the tune parameter (libaom has added -tune vmaf)? not for now

**Question10:**
- Anything missing from the encoder features ? only 10-bit palette is missing from the BDR features, the rest are a WIP such as (s-frames, reference scaling ... )

**Question11:**
- any updates on the ffmpeg integration?
need to finish the api cleanup first

**Comment by luigi311** [comment]
Is there a recording of this months meeting similar to the one we had last time?

**Comment by hassount** [comment]
> Is there a recording of this months meeting similar to the one we had last time?

unfortunately I could not record on my side. @mafonsoN did you get the chance to record the meeting ?

**Comment by mafonsoN** [comment]
@luigi311 @hassount No, sorry, I would have to have to enabled it on my side. We will make sure to record and publish it next time! @hassount, do we want to publish some meeting notes instead from what we remember? I can't seem to get back the chat though.

**Comment by 1480c1** [comment]
@hassount would we be able to host the next dev meeting on another platform, preferably one that has a way to keep a record easily of both text and audio?

**Comment by hassount** [comment]
> @hassount would we be able to host the next dev meeting on another platform, preferably one that has a way to keep a record easily of both text and audio?

sure, any suggestions?

**Comment by 1480c1** [comment]
> > @hassount would we be able to host the next dev meeting on another platform, preferably one that has a way to keep a record easily of both text and audio?
> 
> sure, any suggestions?

Currently researching some, but one that came to mind was a temporary discord server with a [bot](https://craig.chat/home/) to record the audio. I haven't tested the bot yet, but I will soon once I have the time

**Comment by jryzkns** [comment]
> Currently researching some, but one that came to mind was a temporary discord server with a [bot](https://craig.chat/home/) to record the audio. I haven't tested the bot yet, but I will soon once I have the time.

+1 to discord server with bot add-on

**Comment by luigi311** [comment]
Is there a date set for the next meeting? Did we finally decided on just using discord?

**Comment by hassount** [comment]
Let's do February 12th 2020, @ 8:30 - 9:30 am PST

**Comment by EwoutH** [comment]
> Let's do February 12th 2020, @ 8:30 - 9:30 am PST

@hassount Can you open a new issue, or change the title and first message of this issue, to make this stand out a bit more?

**Comment by hassount** [comment]
#1030
